website/blog/2020-03-24-scalacheck.md

```diff
@@ -0,0 +1,96 @@
+---
+author: Gabriele Petronella
+title: Using ScalaCheck with MUnit
+authorURL: https://twitter.com/gabro27
+authorImageURL: https://github.com/gabro.png
+---
+
+Property-based testing is a very popular testing style and its most popular
+implementation for Scala is the [ScalaCheck](https://www.scalacheck.org)
+library.
+
+Starting with version 0.7.0, MUnit introduces a dedicated integration for
+ScalaCheck which we'll explore in this blog post.
+
+<!-- truncate -->
+
+Once you've
+[setup the `munit-scalacheck` module](http://localhost:3001/munit/docs/integrations/scalacheck.html)
+you can extend the `ScalaCheckSuite` trait and start writing your properties,
+for example:
+
+```scala
+import munit.ScalaCheckSuite
+import org.scalacheck.Prop._
+
+class IntegerSuite extends ScalaCheckSuite {
+
+  property("addition is commutative") {
+    forAll { (n1: Int, n2: Int) =>
+      n1 + n2 == n2 + n1
+    }
+  }
+
+  property("0 is the identity of addition") {
+    forAll { (n: Int) =>
+      n + 0 == n
+    }
+  }
+
+}
+```
+
+You can define properties using the familiar features of ScalaCheck, including
+labels (`:|` and `|:`), conditional properties (`==>`) and custom generators.
+
+The integration with MUnit also grants you the ability to use
+[assertions](/munit/docs/assertions.html). This is especially useful when
+testing multiple conditions of a property.
+
+For example, using the ScalaCheck API you may write:
+
+```scala
+property("integer identities") {
+  forAll { (n: Int) =>
+    (n + 0 == n) :| "0 is the addition identity" &&
+    (n * 1 == n) :| "1 is the multiplication identity"
+  }
+}
+```
+
+This is a boolean expression, which uses labels to better identify which part of
+the expression fails.
+
+For longer expressions, however, this may become inconvenient and may choose to
+use MUnit assertions instead:
+
+```scala
+property("integer identities") {
+  forAll { (n: Int) =>
+    assertEquals(n + 0, n)
```

Comment:
Can we include a screenshot of the stdout here? Would be good to use a non-vscode terminal that supports the ANSI highlight color like iTerm.

Reply:
I think it's OK to include the explanation here as well so that it's equivalent to the scalacheck boolean example
```scala
     assertEquals(n + 0, n, "0 is the addition identity")
```